src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -1837,35 +1831,110 @@ internal virtual void AggregateRenderPackages(IEnumerable<HelixRenderPackage> pa
                         { continue;}
 
                         //Otherwise, clean up the remaining mesh geometry data in the render package to exclude the regions already generated.
-                        //First sort the range data
-                        var sortedVerticesRange = new List<Tuple<int, int>>(rp.MeshVerticesRangesAssociatedWithTextureMaps);
-                        sortedVerticesRange.Sort();
-                        sortedVerticesRange.Reverse();
+                        var verticesRange =
+                            new List<Tuple<int, int>>(rp.MeshVerticesRangesAssociatedWithTextureMaps);
                         
-                        //Remove already generated mesh geometry from render package
-                        foreach (var range in sortedVerticesRange)
+                        RemoveMeshGeometryByRange(verticesRange, m);
+                    }
+
+                    //If we are using IRenderInstances data then we need to create a unique Geometry3D object for each instancable item and add instance transforms.  
+                    //If we any mesh geometry was not associated with an instance, remove the previously added mesh data from the render package so the remaining mesh can be added to the scene.
+                    if (rp.MeshVerticesRangesAssociatedWithInstancing.Any())
+                    {
+                        //For each range of mesh vertices add the mesh data and instances to the scene
+                        var meshVertexCountTotal = 0;
+                        var j = 0;
+                        foreach (var item in rp.MeshVerticesRangesAssociatedWithInstancing)
```

Comment:
It will be clearer if `item` is renamed to `range` or something more meaningful to describe what it represents?